src/main/java/com/cloudbees/jenkins/support/api/ObjectComponent.java

```diff
@@ -86,8 +88,12 @@ public static <T extends AbstractModelObject> List<ObjectComponentDescriptor<T>>
      */
     @Override
     public <C extends AbstractModelObject> boolean isApplicable(Class<C> clazz) {
-        Class<T> type = (Class<T>) ((ParameterizedType) getClass().getGenericSuperclass()).getActualTypeArguments()[0];
-        return type.isAssignableFrom(clazz);
+        Type parameterization = Types.getBaseClass(getClass(), ObjectComponent.class);
+        if (parameterization instanceof ParameterizedType pt) {
+            return Types.erasure(Types.getTypeArgument(pt, 0)).isAssignableFrom(clazz);
```

Comment:
Not sure what this change is going, but FWIW if you need to use `org.jvnet.tiger_types` you have already lost any sort of debuggability. Just make subclasses explicitly implement `isApplicable`.

Reply:
> Looking in core, I found out that we use `org.jvnet.tiger_types` in such kind of cases.

Yes, and we cannot fix that because we would be forced to break compatibility of poorly designed APIs used by plugins. There should be no such constraint here.